e2etest/l2tests/assignment.go

```diff
@@ -515,4 +518,57 @@ var _ = ginkgo.Describe("IP Assignment", func() {
 			framework.ExpectNoError(err)
 		})
 	})
+
+	ginkgo.Context("IPV4 - assign specifc IP for LB service", func() {
+		ginkgo.It("should preseve the same external ip after controller restart", func() {
+			specificIP := "192.168.10.100"
+
+			resources := config.Resources{
+				Pools: []metallbv1beta1.IPAddressPool{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "assignment-controller-reset-test-pool",
+						},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: []string{"192.168.10.100/32"},
```

Comment:
Can we have two ips and ensure that a service with an assigned ip gets the same ip after the restart (which should be the case, given the new logic)

Reply:
Yes. done.